### PR TITLE
Pull Request for branch addreducersandactions that adds Reducers and actions

### PR DESCRIPTION
**Pull request by jkanyi-web**
- Added file redux/books/booksSlice.js that defines a Redux slice called books that manages an array of books. It has two reducers for adding and removing books, and exports action creators for these reducers. It also has an initialState which is an object with a List property set to an empty array.
- Also added file redux/categories/categoriesSlice.js that manages a list of categories, with a single reducer function called checkStatus that updates the List property of the state object based on the value of the action.payload. It exports an action creator and a reducer function that can be used with the createStore() function from Redux.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { createSlice } from '@reduxjs/toolkit';
+import List from '../../components/List';
+
+const initialState = {
+  List,
+};
```
Comment (mckent05) [inline src/redux/books/booksSlice.js]: - You have done well so far,  but I noticed you are using a component as your initial state, this is not considered good practice, according to the project requirements, you are to create an `initialState` with a state value `books` that contains an empty array. Please fix this 🙏

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          return {
+            ...category,
+            status: statusToUpdate,
+          };
+        }
+        return category;
+      });
+    },
+  },
+});
```
Comment (mckent05) [inline src/redux/categories/categoriesSlice.js]: - You have done well so far, but the logic for your checkStatus action would return an error when the list in the initialState is empty as the value of category would be null. I suggest you consider using the code below:
```javascript
export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})
```

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @jkanyi-web, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- You have done well adding reducers and actions to your application 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — APPROVED** [review]
Hi @jkanyi-web 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

https://github.com/jkanyi-web/bookstore-redo-cra/blob/2ad48760b778617741d896c3eaed4f967b0579f1/README.md?plain=1#L56-L62

- [ ]  [OPTIONAL] Please remove these lines, as we don't have `Live Demo and video ` in this case, or you may put `N/A` to make your README more professional 👍 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**